examples/federated/core_package/index.js

```diff
@@ -48,29 +50,29 @@ async function main() {
   const extension_data = JSON.parse(
     PageConfig.getOption('dynamic_extensions')
   );
-  const mime_extension_data = JSON.parse(
-    PageConfig.getOption('dynamic_mime_extensions')
-  );
+
+  const dynamicPlugins = [];
+  const dynamicMimePlugins = [];
+  const dynamicStyles = [];
 
   // Get dynamic plugins
   // TODO: deconflict these with builtins?
-  const dynamicPromises = extension_data.map(data =>
-    loadComponent(
-      data.path,
-      data.name,
-      data.module
-    )
-  );
-  const dynamicPlugins = await Promise.all(dynamicPromises);
-
-  const dynamicMimePromises = mime_extension_data.map(data =>
-    loadComponent(
-      data.path,
-      data.name,
-      data.module
+  const componentLoadPromises = extension_data.map(async (data) => {
+    await loadComponent(
+      `${URLExt.join(PageConfig.getOption('fullLabextensionsUrl'), data.name, 'remoteEntry.js')}`,
+      data.name
     )
-  );
-  const dynamicMimePlugins = await Promise.all(dynamicMimePromises);
+    if (data.plugin) {
+      dynamicPlugins.push(await createModule(data.name, data.plugin));
```

Comment:
These three promises are in series here, but should be in parallel. I think it would be better to work with the actual promise returned by loadComponent, rather than using async/await, so that the promise resolution could be done in parallel.